[gdal] build failure with libkml without expat feature

<!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
<!-- ⁉ You can find its location at the end of the console output. ⁉ -->

**Host Environment**

- OS: Windows
- Compiler: MSVC 17.4.4

**To Reproduce**

Steps to reproduce the behavior:

`./vcpkg install 'gdal[core,libkml]:x64-windows'`

**Failure logs**

```
Computing installation plan...
The following packages will be built and installed:
    gdal[core,libkml]:x64-windows -> 3.6.2#2
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\Russell\AppData\Local\vcpkg\archives in 168.6 us. Use --debug to see more details.
Installing 1/1 gdal:x64-windows...
Building gdal[core,libkml]:x64-windows...
-- Using cached OSGeo-gdal-v3.6.2.tar.gz.
-- Cleaning sources at C:/Users/Russell/Documents/vcpkg/buildtrees/gdal/src/v3.6.2-7ba5134fbb.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/Russell/Documents/vcpkg/downloads/OSGeo-gdal-v3.6.2.tar.gz
-- Applying patch find-link-libraries.patch
-- Applying patch fix-gdal-target-interfaces.patch
-- Applying patch fix-find-package2.patch
-- Applying patch libkml.patch
-- Using source at C:/Users/Russell/Documents/vcpkg/buildtrees/gdal/src/v3.6.2-7ba5134fbb.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j17
    Working Directory: C:/Users/Russell/Documents/vcpkg/buildtrees/gdal/x64-windows-dbg
    See logs for more information:
      C:\Users\Russell\Documents\vcpkg\buildtrees\gdal\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/gdal/portfile.cmake:100 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building gdal:x64-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+gdal
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[gdal]+Build+error
Include '[gdal] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-12-14-7ae0d8527fb488fde10a89c2813802dc9b03b6f9
    vcpkg-scripts version: cf69bc743 2023-02-08 (65 minutes ago)


Please use the prefilled template from C:\Users\Russell\Documents\vcpkg\installed\vcpkg\issue_body.md when reporting your issue.
```


[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/10690382/install-x64-windows-dbg-out.log)

<!-- ‼ **The console output by itself is not enough to investigate a build failure.** ‼ -->

**Additional context**

It is notable that `.\vcpkg.exe install 'gdal[core,libkml,expat]:x64-windows'` does work perfectaly fine

--- a/ports/libkml/add-target-include-directories.patch
+++ b/ports/libkml/add-target-include-directories.patch
@@ -1,0 +1,12 @@
+diff --git a/cmake/LibKMLHelper.cmake b/cmake/LibKMLHelper.cmake
+index 5cc166a..3ed3079 100644
+--- a/cmake/LibKMLHelper.cmake
++++ b/cmake/LibKMLHelper.cmake
+@@ -1,6 +1,7 @@
+ macro(build_target)
+   cmake_parse_arguments(LIB  "" "NAME" "SRCS;INCS;LINKS;DEPENDS" ${ARGN} )
+   add_library(${LIB_NAME} ${LIB_SRCS})
++  target_include_directories(${LIB_NAME} INTERFACE $<INSTALL_INTERFACE:include>)
+ 
+   foreach(LIB_DEPEND ${LIB_DEPENDS})
+     add_dependencies(${LIB_NAME} ${LIB_DEPEND})

--- a/ports/libkml/portfile.cmake
+++ b/ports/libkml/portfile.cmake
@@ -5,13 +5,14 @@ endif()
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libkml/libkml
-    REF   1.3.0
+    REF  "${VERSION}"
     SHA512 aa48158103d3af764bf98c1fb4cf3e1356b9cc6c8e79d80b96850916f0a8ccb1dac3a46427735dd0bf20647daa047d10e722ac3da2a214d4c1559bf6d5d7c853
     HEAD_REF master
     PATCHES
         patch_empty_literal_on_vc.patch
         fix-mingw.patch
         fix-minizip.patch
+        add-target-include-directories.patch
 )
 
 file(REMOVE
@@ -39,4 +40,4 @@ else()
 endif()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/libkml/vcpkg.json
+++ b/ports/libkml/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libkml",
   "version": "1.3.0",
-  "port-version": 10,
+  "port-version": 11,
   "description": "Reference implementation of OGC KML 2.2",
   "homepage": "https://github.com/libkml/libkml",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4018,7 +4018,7 @@
     },
     "libkml": {
       "baseline": "1.3.0",
-      "port-version": 10
+      "port-version": 11
     },
     "liblas": {
       "baseline": "1.8.1",

--- a/versions/l-/libkml.json
+++ b/versions/l-/libkml.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "4969a3e81df3ef13b5e3f2f94e652311df8bbafe",
+      "version": "1.3.0",
+      "port-version": 11
+    },
+    {
       "git-tree": "8e1c949e666db286674cb1af98edb4d23d7ec843",
       "version": "1.3.0",
       "port-version": 10

